Documentation: Add and update documentation for metal variants (#2205)
bottlerocket-os/bottlerocket: SUPPORTED-HARDWARE.md

# Supported hardware for Bottlerocket on bare metal

This document captures the hardware Bottlerocket for bare metal is tested on and known to work with.

> It also contains hardware the community has reported to be functional.

Reviewer: nit:
```suggestion
It also contains hardware that contributors have reported to be functional.
```
